### PR TITLE
[WIP] Feature : Adding clip command to CLI

### PR DESCRIPTION
**Pull request by louib**
## Description
Adding a command to the CLI interface to copy a password to the clipboard.

I still want to allow for specifying an entry by name, using the `EntrySearcher`, before this gets merged.

## How has this been tested?
Unit tests + local testing.

## Types of changes

- ✅ New feature

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/cli/Clip.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    out << "Insert the database password\n> ";
+    out.flush();
+
+    static QTextStream inputTextStream(stdin, QIODevice::ReadOnly);
+    QString line = inputTextStream.readLine();
+    CompositeKey key = CompositeKey::readFromLine(line);
+
+    Database* db = Database::openDatabaseFile(args.at(0), key);
+    if (db == nullptr) {
```
Comment (phoerious) [inline src/cli/Clip.cpp]: !db

File: src/cli/Clip.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    QString entryId = args.at(1);
+    if (!Uuid::isUuid(entryId)) {
+        qCritical("Invalid Uuid %s.", qPrintable(entryId));
+        return EXIT_FAILURE;
+    }
+
+    Uuid entryUuid = Uuid::fromHex(entryId);
+    Entry* entry = db->resolveEntry(entryUuid);
+    if (entry == nullptr) {
```
Comment (phoerious) [inline src/cli/Clip.cpp]: same

File: src/cli/Clip.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        qCritical("No entry found with Uuid %s.", qPrintable(entryId));
```
Comment (phoerious) [inline src/cli/Clip.cpp]: Maybe write UUID? Uuid looks ugly.

**Comment by louib** [comment]
I still have a lot of work to do on this PR. I want to be able to test the CLI commands, and right now it's very hard to do. I'll do the required refactoring and add tests for the current commands before reopening the pull request.

**Pull request closed without merging**